pandas-stubs/core/tools/numeric.pyi

```diff
@@ -1 +1,64 @@
-def to_numeric(arg, errors: str = ..., downcast=...): ...
+from typing import (
+    Literal,
+    overload,
+)
+
+import numpy as np
+import pandas as pd
+
+from pandas._typing import (
+    IgnoreRaiseCoerce,
+    Scalar,
+    npt,
+)
+
+@overload
+def to_numeric(
+    arg: Scalar,
+    errors: Literal["raise", "coerce"] = ...,
+    downcast: Literal["integer", "signed", "unsigned", "float"] | None = ...,
+) -> int | float: ...
+@overload
+def to_numeric(
+    arg: Scalar,
+    errors: Literal["ignore"],
+    downcast: Literal["integer", "signed", "unsigned", "float"] | None = ...,
+) -> Scalar: ...
+@overload
+def to_numeric(
+    arg: list | tuple | npt.NDArray,
+    errors: IgnoreRaiseCoerce = ...,
+    *,
+    downcast: Literal["integer", "signed", "unsigned", "float"],
+) -> npt.NDArray: ...
+@overload
+def to_numeric(
+    arg: list | tuple | npt.NDArray,
+    errors: Literal["ignore"],
+    downcast: None = ...,
+) -> npt.NDArray: ...
```

Comment:
aren't these 2 overloads really the same?  They both return `npt.NDArray`. 

Based on the docs, if you use `errors = "ignore"`, then you get back the input (so maybe it's the same type)?

Reply:
Nearly the same, but the arguments differ to get the untyped. No complaints form mypy for overlap, and both are tested. 

Docs seem to be wrong:

```python
In [2]: pd.to_numeric([1,2,3],errors="ignore")
Out[2]: array([1, 2, 3], dtype=int64)

In [3]: pd.to_numeric([1,2,3,"none"],errors="ignore")
Out[3]: array([1, 2, 3, 'none'], dtype=object)
```


In either case, if you ignore and it works, then you get an intp or float array (w/o further args set).  If you hit a bad value, you seem to always get object.  In either case, can't tell which will happen statically.

Reply:
Maybe I'm missing something.  Why not do:
```python
def to_numeric(
    arg: list | tuple | npt.NDArray,
    errors: IgnoreRaiseCoerce = ...,
    downcast: Literal["integer", "signed", "unsigned", "float"] | None = ...
) -> npt.NDArray: ...
```

Reply:
Also simplified.